TimeTrackerManager/TimeTrackerManager/Core/Infra/FirebaseUsersLoader.swift

```diff
@@ -0,0 +1,77 @@
+//
+//  UserLoader.swift
+//  TimeTrackerManager
+//
+//  Created by Bocanu Mihai on 11.01.2022.
+//
+
+import Foundation
+import Firebase
+import TimeTrackerCore
+import SwiftUI
+import UIKit
+import WebKit
+
+class FirebaseUsersLoader  {
+
+    init(store: TimeslotsStore){
+        self.store = store
+    }
+
+    struct UndefinedError: Error { }
+    typealias GetUsersResult = (Result<[UserCell], Error>) -> Void
+    typealias GetUserInfoResult = (Result<Void, Error>) -> Void
+    typealias GetTimeslotsResult = (Result<[TimeSlot], Error>) -> Void
+    var store: TimeslotsStore
+
+
+    func getUsers(completion: @escaping GetUsersResult) {
+        Firestore.firestore().collection(Path.users).getDocuments() { (snapshot, error) in
+            if let snapshot = snapshot {
+                let users = snapshot.documents.compactMap { document -> UserCell? in
+                    let data = document.data()
+                    let userId = data["userId"] as? String ?? ""
+                    var totalHours = 0
+                    var allProjects = Set<String>()
+
+                   self.getUserTimeslots(userId) { result in
+                        if case let .success(result) = result {
+                            result.forEach { timeSlot in
+                                totalHours += timeSlot.total
+                                allProjects.insert(timeSlot.projectName)
+                            }
+                        }
+                        if case let .failure(error) = result {
+                            print("ERROR", error)
```

Comment:
trebuie sa ne gandim si aici ce facem in caz de failure. Eventual poti pune un TODO momentan sa nu uitam